jira/client.py

```diff
@@ -3438,9 +3446,9 @@ def backup_download(self, filename=None):
             logging.error(ioe)
         return None
 
-    def current_user(self, field="key"):
-        """Returns the username or emailAddress of the current user. For anonymous
-        users it will return a value that evaluates as False.
+    def current_user(self, field="accountId"):
```

Comment:
This change is not backwards compatible and will break non-cloud versions of Jira.

Reply:
Hmm, I noticed that similar BC breaking changes have already been committed to master (e.g. a4e6b1b2). Not too fond of this, but I guess restoring the BC could better be left for a seperate PR.

Reply:
The CI tests are only against the cloud version, which seems to be the priority for this project. Given the limited resources available to maintain it, I would tend to agree with breaking older on-prem versions in favour of keeping track with the cloud and maintaining BC on a best-effort basis.

The GDPR compliance changes also represent breaking changes within a major version of the API. This represents a fair amount of work to restore functionality to the library as more things get deprecated and dropped. As far as I can tell it also has caused a fair amount of divergence between on-prem and cloud versions, increasing the difficulty of maintenance.

With these factors in mind I'd prefer to prioritize getting fixes in to bring CI back to green and raise issues that arise afterwards, so that the project can gain momentum again. With CI green, it should be easier to put forward smaller PRs that are easier to merge to fix BC piece-by-piece.

Reply:
I agree that getting the CI green again should be a top priority. Over the weekend I was working on getting the CI to run against a Jira instance spinned up as part of the pipeline, as to get rid of the credential management issue and have more reproducible builds. So in that effort fixing CI and maintaining BC align.

Anyway, this was just the first BC breaking change I ran into. Shouldn't be considered a blocker for this PR.